modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -1194,6 +1194,8 @@ private ObjectField _addObjectField(
 		objectField.setRelationshipType(relationshipType);
 		objectField.setRequired(required);
 
+		_validateRelationshipNameForDBColumnLimit(name, objectDefinition1);
```

Comment:
Can you move this validation up? We often validate the values before set them in the entity.

I think this should be right before `if (Validator.isNull(dbColumnName)) {`